src/openepd/m49/geo_converter.py

```diff
@@ -0,0 +1,179 @@
+#
+#  Copyright 2025 by C Change Labs Inc. www.c-change-labs.com
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+from openepd.m49.const import (
+    COUNTRY_VERBOSE_NAME_TO_M49,
+    ISO3166_ALPHA2_TO_M49,
+    M49_TO_COUNTRY_VERBOSE_NAME,
+    M49_TO_ISO3166_ALPHA2,
+    M49_TO_REGION_VERBOSE_NAME,
+    OPENEPD_SPECIAL_REGIONS,
+    REGION_VERBOSE_NAME_TO_M49,
+)
+
+
+def iso_to_m49(regions: list[str]) -> set[str]:
+    """
+    Convert ISO3166 alpha2 country codes to M49 region codes.
+
+    :param:
+        regions: List of ISO3166 alpha2 country codes (e.g., ["US", "CA", "MX"])
+
+    :return:
+        Set of M49 region codes (e.g., ["840", "124", "484"])
+    """
+
+    if not regions:
+        raise ValueError("No data provided.")
+
+    result = set()
+    for code in regions:
+        m49_code = ISO3166_ALPHA2_TO_M49.get(code.upper())
+        if m49_code:
+            result.add(m49_code)
+        else:
+            raise ValueError(f"Country code '{code}' not found in M49 region codes.")
+
+    return result
+
+
+def m49_to_iso(regions: list[str]) -> set[str]:
+    """
+    Convert M49 region codes to ISO3166 alpha2 country codes.
+
+    :param:
+        regions: List of M49 region codes (e.g., ["840", "124", "484"])
+
+    :return:
+        Set of ISO3166 alpha2 country codes (e.g., ["US", "CA", "MX"])
+    """
+
+    if not regions:
+        raise ValueError("No data provided.")
+
+    result = set()
+    for code in regions:
+        iso_code = M49_TO_ISO3166_ALPHA2.get(code.upper())
+        if iso_code:
+            result.add(iso_code)
+        else:
+            raise ValueError(f"Region code '{code}' not found in ISO3166.")
+
+    return result
+
+
+def region_and_country_names_to_m49(regions: list[str]) -> set[str]:
+    """
+    Convert user-friendly region and country names to M49 region codes.
+
+    :param:
+        regions: List of user-friendly region and country names (e.g., ["Europe", "North America", "Austria", "Germany"])
+
+    :return:
+        Set of M49 region codes (e.g., ["150", "003", "040", "276"])
+    """
+
+    if not regions:
+        raise ValueError("No data provided.")
+
+    result = set()
+    for name in regions:
+        m49_code = REGION_VERBOSE_NAME_TO_M49.get(name.title()) or COUNTRY_VERBOSE_NAME_TO_M49.get(name.title())
```

Comment:
`assert region_and_country_names_to_m49("Bolivia") == "068"` - is this a valid case? If yes then it will not work.